nncf/torch/quantization/layers.py

```diff
@@ -1039,14 +1039,18 @@ class WeightsDecompressor(nn.Module):
     Applies decompression of compressed weights in the forward pass
     """
 
-    def __init__(self, scale: torch.Tensor, zero_point: torch.Tensor):
+    def __init__(self, scale: torch.Tensor, zero_point: torch.Tensor, result_dtype: torch.dtype = None):
```

Comment:
Could you explain why `result_dtype` is optional?

Reply:
The idea was to also support the case when no explicit casting is needed